pkg/health/health_pod.go

```diff
@@ -117,14 +121,24 @@ func getCorev1PodHealth(pod *corev1.Pod) (*HealthStatus, error) {
 				Status:  HealthStatusProgressing,
 				Message: pod.Status.Message,
 			}, nil
-		case corev1.RestartPolicyOnFailure, corev1.RestartPolicyNever:
-			// pods set with a restart policy of OnFailure or Never, have a finite life.
-			// These pods are typically resource hooks. Thus, we consider these as Progressing
-			// instead of healthy.
-			return &HealthStatus{
-				Status:  HealthStatusProgressing,
-				Message: pod.Status.Message,
-			}, nil
+		}
+		if _, hook := pod.Annotations[AnnotationIgnoreRestartPolicy]; hook {
+			return getHealthStatus(pod)
+		} else {
+			switch pod.Spec.RestartPolicy {
+			case corev1.RestartPolicyAlways:
+				return getHealthStatus(pod)
+			case corev1.RestartPolicyOnFailure, corev1.RestartPolicyNever:
+				// Most pods set with a restart policy of OnFailure or Never, have a finite life.
+				// These pods are typically resource hooks. Thus, we consider these as Progressing
+				// instead of healthy. If this is unwanted, e.g., when the pod is managed by an
+				// operator and therefore has a restart policy of OnFailure or Never, then use the
+				// the AnnotationIgnoreRestartPolicy annotation.
+				return &HealthStatus{
+					Status:  HealthStatusProgressing,
+					Message: pod.Status.Message,
+				}, nil
+			}
```

Comment:
```suggestion
                policy := pod.Spec.RestartPolicy
		if _, ok := pod.Annotations[AnnotationIgnoreRestartPolicy]; ok || policy == corev1.RestartPolicyAlways {
			return getHealthStatus(pod)
		}
		
			 if policy == corev1.RestartPolicyOnFailure || policy == corev1.RestartPolicyNever {
				// Most pods set with a restart policy of OnFailure or Never, have a finite life.
				// These pods are typically resource hooks. Thus, we consider these as Progressing
				// instead of healthy. If this is unwanted, e.g., when the pod is managed by an
				// operator and therefore has a restart policy of OnFailure or Never, then use the
				// the AnnotationIgnoreRestartPolicy annotation.
				return &HealthStatus{
					Status:  HealthStatusProgressing,
					Message: pod.Status.Message,
				}, nil
			}
```

I think it's little bit of a easy to read.